MapboxNavigation/NavigationMapView.swift

```diff
@@ -1350,3 +1357,114 @@ extension NavigationMapView {
+            return
+        }
+        
+        addStyleLayersIfNecessary() // may need to re-add the layers if the map style has changed (e.g. preview style -> nav style)
+        
+        if let highlightedBuildingFillExtrusionLayer = style?.layer(withIdentifier: StyleLayerIdentifier.buildingExtrusion) as? MGLFillExtrusionStyleLayer {
+            var predicate = "extrude = 'true' && type = 'building' && underground = 'false'"
+            if extrudeAll == false {
+                // form a predicate to filter out the other buildings from the datasource so only the desired ones are included
+                var identifiersBoolean = "($featureIdentifier == \(buildings.first!.0)"
+                
+                if buildings.count > 1 {
+                    for building in buildings[1...buildings.count-1] {
+                        identifiersBoolean += " || $featureIdentifier == \(building.0)"
+                    }
+                }
+                
+                identifiersBoolean += ")"
+                
+                predicate += " && \(identifiersBoolean)"
+            }
+            
+            highlightedBuildingFillExtrusionLayer.predicate = NSPredicate(format: predicate)
+            
+            // we have some specific building IDs that we will be the highlight color
+            // the rest of the buildings will be extruded but kept at a uniform color
+            var colorsList = [UIColor]()
+            var highlightedBuildingHeightExpression = "MGL_MATCH($featureIdentifier, "
+            var highlightedBuildingColorExpression = "MGL_MATCH($featureIdentifier, "
+            
+            for building in buildings {
+                highlightedBuildingHeightExpression += "\(building.0), height, "
+                highlightedBuildingColorExpression += "\(building.0), %@, "
+                colorsList.append(building.1)
+            }
+            
+            highlightedBuildingColorExpression += "%@)"
+            
+            // TODO: Verify whether extrusion only for specific building or all buildings is required.
+            colorsList.append(.green)
+            
+            if extrudeAll == true {
+                highlightedBuildingHeightExpression += "height)"
+            } else {
+                highlightedBuildingHeightExpression += "0)"
+            }
+            
+            let fillExtrusionHeightStops = [0: NSExpression(forConstantValue: 0),
+                                            15: NSExpression(forConstantValue: 0),
+                                            15.25: NSExpression(format: highlightedBuildingHeightExpression)]
+            
+            let fillExtrusionBaseStops = [0: NSExpression(forConstantValue: 0),
+                                          15: NSExpression(forConstantValue: 0),
+                                          15.25: NSExpression(forKeyPath: "min_height")]
+            
+            let opacityStops = [15: 0.5, 17: 0.8]
```

Comment:
According to these stops, the buildings would be completely flat until z15, then grow to a realistic height by z15.25. Is the zoom level at the time of arrival guaranteed to be at least z15? Do we need it to depend on the zoom level?

Reply:
I'm not author of initial implementation, but seems that we can safely skip first gradient stop. @avi-c, do you maybe remember what was the original idea?
```suggestion
            let fillExtrusionHeightStops = [15: NSExpression(forConstantValue: 0),
                                            15.25: NSExpression(format: highlightedBuildingHeightExpression)]
            
            let fillExtrusionBaseStops = [15: NSExpression(forConstantValue: 0),
                                          15.25: NSExpression(forKeyPath: "min_height")]
            
            let opacityStops = [15: 0.5, 17: 0.8]
```